expression/builtin_arithmetic_vec.go

```diff
@@ -820,115 +820,166 @@ func (b *builtinArithmeticIntDivideIntSig) vectorized() bool {
 func (b *builtinArithmeticIntDivideIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
 	return errors.Errorf("not implemented")
 }
-
-func (b *builtinArithmeticPlusIntSig) vectorized() bool {
+func (b *builtinArithmeticPlusIntUnsignedUnsignedSig) vectorized() bool {
 	return true
 }
-
-func (b *builtinArithmeticPlusIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+func (b *builtinArithmeticPlusIntUnsignedUnsignedSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
 	n := input.NumRows()
 	lh, err := b.bufAllocator.get(types.ETInt, n)
 	if err != nil {
 		return err
 	}
 	defer b.bufAllocator.put(lh)
-
 	if err := b.args[0].VecEvalInt(b.ctx, input, lh); err != nil {
 		return err
 	}
-
 	// reuse result as rh to avoid buf allocate
 	if err := b.args[1].VecEvalInt(b.ctx, input, result); err != nil {
 		return err
 	}
-
 	result.MergeNulls(lh)
-
 	rh := result
 	lhi64s := lh.Int64s()
 	rhi64s := rh.Int64s()
 	resulti64s := result.Int64s()
-
-	isLHSUnsigned := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
-	isRHSUnsigned := mysql.HasUnsignedFlag(b.args[1].GetType().Flag)
-
-	switch {
-	case isLHSUnsigned && isRHSUnsigned:
-		err = b.plusUU(result, lhi64s, rhi64s, resulti64s)
-	case isLHSUnsigned && !isRHSUnsigned:
-		err = b.plusUS(result, lhi64s, rhi64s, resulti64s)
-	case !isLHSUnsigned && isRHSUnsigned:
-		err = b.plusSU(result, lhi64s, rhi64s, resulti64s)
-	case !isLHSUnsigned && !isRHSUnsigned:
-		err = b.plusSS(result, lhi64s, rhi64s, resulti64s)
-	}
+	err = b.plusUU(result, lhi64s, rhi64s, resulti64s)
 	return err
 }
-func (b *builtinArithmeticPlusIntSig) plusUU(result *chunk.Column, lhi64s, rhi64s, resulti64s []int64) error {
+func (b *builtinArithmeticPlusIntUnsignedUnsignedSig) plusUU(result *chunk.Column, lhi64s, rhi64s, resulti64s []int64) error {
 	for i := 0; i < len(lhi64s); i++ {
 		if result.IsNull(i) {
 			continue
 		}
 		lh, rh := lhi64s[i], rhi64s[i]
-
-		if uint64(lh) > math.MaxUint64-uint64(rh) {
-			return types.ErrOverflow.GenWithStackByArgs("BIGINT UNSIGNED", fmt.Sprintf("(%s + %s)", b.args[0].String(), b.args[1].String()))
+		res, _, err := b.plusUnsingedunsinged(lh, rh)
```

Comment:
Do you think we should manually inline the implementation to avoid this function call? @qw4990